src/v4l/v4lt.rs

```diff
@@ -0,0 +1,148 @@
+
+                // Output devices generally cannot know the exact size of the output buffers for
+                // compressed formats (e.g. MJPG). They will however allocate a size that is always
+                // large enough to hold images of the format in question. We know how big a buffer we need
+                // since we control the input buffer - so just enforce that size on the output buffer.
+                let buf_out = &mut buf_out[0..payload.data.len()];
+
+                buf_out.copy_from_slice(&payload.data);
+                buf_out_meta.field = 0;
+                //buf_out_meta.bytesused = buf_in_meta.bytesused;
+            }
+            BcMedia::Pframe(payload) => {
+                let video_type = match payload.video_type {
+                    VideoType::H264 => StreamFormat::H264,
+                    VideoType::H265 => StreamFormat::H265,
+                };
+                self.set_format(Some(video_type));
+                //self.vidsrc.write_all(&payload.data)?;
+
+                let (buf_out, buf_out_meta) = OutputStream::next(self.vidsrc).unwrap();
+
+                // Output devices generally cannot know the exact size of the output buffers for
+                // compressed formats (e.g. MJPG). They will however allocate a size that is always
+                // large enough to hold images of the format in question. We know how big a buffer we need
+                // since we control the input buffer - so just enforce that size on the output buffer.
+                let buf_out = &mut buf_out[0..payload.data.len()];
+
+                buf_out.copy_from_slice(&payload.data);
+                buf_out_meta.field = 0;
+                //buf_out_meta.bytesused = buf_in_meta.bytesused;
+            }
+            _ => {
+                //Ignore other BcMedia
+            }
+        }
+
+        Ok(())
+    }
+}
+
+impl<'b, 'c> V4ltOutputs<'b, 'c> {
+    pub(crate) fn from_appsrcs(device: &'c mut Device, vidsrc: &'b mut MmapStream<'b>) -> V4ltOutputs<'b, 'c> {
+        let result = V4ltOutputs {
+            vidsrc,
+            video_format: None,
+            device,
+        };
+        result.apply_format();
+        result
+    }
+
+    fn set_format(&mut self, format: Option<StreamFormat>) {
+        match format {
+            Some(StreamFormat::H264) | Some(StreamFormat::H265) => {
+                if format != self.video_format {
+                    self.video_format = format;
+                    self.apply_format();
+                }
+            }
+            _ => {}
+        }
+    }
+
+    fn apply_format(&self) {
+        let launch_vid = match self.video_format {
+            Some(StreamFormat::H264) => {
+                b"AVC1"
+            }
+            Some(StreamFormat::H265) => {
+                b"HEVC"
+            }
+            // TODO
+            _ => b"AVC1",
+        };
+        // TODO
+        let fmt = Format::new(640, 480, FourCC::new(launch_vid));
```

Comment:
If you listen to the `BcMediaInfoV2` and `BcMediaInfoV1` packets as suggested above you can set the sizes using that